fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4

```diff
@@ -1766,6 +1767,23 @@ complexColType
     : identifier COLON dataType commentSpec?
     ;
 
+variantTypeDefinitions
+    : VARIANT LT variantSubColTypeList COMMA properties=propertyClause GT  #variantWithFieldsAndProps
+    | VARIANT LT variantSubColTypeList GT                                  #variantWithOnlyFields
+    | VARIANT LT properties=propertyClause GT                              #variantWithOnlyProps
+    | VARIANT                                                              #variant
```

Comment:
use same alias for them, then u could process them in one function in LogicalPlanBuilder